prowler/providers/iac/iac_provider.py

```diff
@@ -74,6 +74,68 @@ def setup_session(self):
         """IAC provider doesn't need a session since it uses Checkov directly"""
         return None
 
+    def _process_check(self, finding: dict, check: dict, status: str) -> CheckReportIAC:
+        """
+        Process a single check (failed or passed) and create a CheckReportIAC object.
+
+        Args:
+            finding: The finding object from Checkov output
+            check: The individual check data (failed_check or passed_check)
+            status: The status of the check ("FAIL" or "PASS")
+
+        Returns:
+            CheckReportIAC: The processed check report
+        """
+        # Get severity and ensure it's a valid string
+        severity = finding.get("severity", "low")
```

Comment:
```suggestion
        severity = check.get("severity", "low")
```

severity should come from each, not the finding (finding here is still all the findings for any check_type)